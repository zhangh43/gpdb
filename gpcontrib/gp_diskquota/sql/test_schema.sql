-- Test schema
create schema s1;
select diskquota.set_schema_quota('s1', '1 MB');
set search_path to s1;

create table a(i int);
insert into a select generate_series(1,100);
-- expect insert fail
insert into a select generate_series(1,100000000);
-- expect insert fail
insert into a select generate_series(1,100);
create table a2(i int);
-- expect insert fail
insert into a2 select generate_series(1,100);

-- Test alter table set schema
create schema s2;
alter table s1.a set schema s2;
select pg_sleep(5);
-- expect insert succeed
insert into a2 select generate_series(1,20000);
-- expect insert succeed
insert into s2.a select generate_series(1,20000);

alter table s2.a set schema badquota;
-- expect failed
insert into badquota.a select generate_series(0, 100);

reset search_path;
drop table s1.a2, badquota.a;
drop schema s1, s2;

