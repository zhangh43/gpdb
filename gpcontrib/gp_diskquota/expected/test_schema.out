-- Test schema
create schema s1;
select diskquota.set_schema_quota('s1', '1 MB');
 set_schema_quota 
------------------
 
(1 row)

set search_path to s1;
create table a(i int);
insert into a select generate_series(1,100);
-- expect insert fail
insert into a select generate_series(1,100000000);
ERROR:  schema's disk space quota exceeded with name:s1
-- expect insert fail
insert into a select generate_series(1,100);
ERROR:  schema's disk space quota exceeded with name:s1
create table a2(i int);
-- expect insert fail
insert into a2 select generate_series(1,100);
ERROR:  schema's disk space quota exceeded with name:s1
-- Test alter table set schema
create schema s2;
alter table s1.a set schema s2;
select pg_sleep(5);
 pg_sleep 
----------
 
(1 row)

-- expect insert succeed
insert into a2 select generate_series(1,20000);
-- expect insert succeed
insert into s2.a select generate_series(1,20000);
alter table s2.a set schema badquota;
-- expect failed
insert into badquota.a select generate_series(0, 100);
ERROR:  schema's disk space quota exceeded with name:badquota
reset search_path;
drop table s1.a2, badquota.a;
drop schema s1, s2;
